AVX-64: Add HTTPS support

dat-tran-matternet opened the pull request.

dat-tran-matternet committed:
AVX-64: Add HTTPS support

bhiln commented on ntripclient.c:
```diff
@@ -566,6 +569,32 @@ static int encode(char *buf, int size, const char *user, const char *pwd)
+
+        // Wait for 100ms before trying again.
+        if (result == CURLE_AGAIN)
+            usleep(100 * 1000);
```
consider defining 1ms to remove magic number:
```
unsigned int one_ms = 1000;
...
usleep(100 * one_ms);
```

dat-tran-matternet replied:
Done. Made a macro for SLEEP_MS.

bhiln commented on ntripclient.c:
```diff
@@ -1432,8 +1461,29 @@ int main(int argc, char **argv)
+          curl_easy_setopt(curl, CURLOPT_URL, url);
+          curl_easy_setopt(curl, CURLOPT_PORT, atoi(port));
+          curl_easy_setopt(curl, CURLOPT_CONNECT_ONLY, 1L);
+          //curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
```
Please remove this line if it is no longer used

dat-tran-matternet replied:
Done. This line is needed time to time to debug the connection during development. I changed the setting to 0. Any developer can change back to 1 for debugging in the future.

bhiln reviewed (approved):
Left minor suggestions, but looks great!

dat-tran-matternet committed:
AVX-64: Handle partial reads of HTTP header

Adding HTTPS causes some overhead that exposed a flaw of the
original code that handles the parsing of the HTTP header resonse.
The original code assumed there weren't any partial messages.
This caused the program to fail in parsing the HTTP header when
using TLS for HTTPS connection.

This change handles for reading of partial header content to
solve the above issue.

rabad-matternet reviewed (approved):
This looks good but please test before merging.

dat-tran-matternet committed:
AVX-64: Send NMEA sentence after HTTP GET message

This change removes the NMEA string from the HTTP GET message.
Instead, the NMEA string is sent after the initial HTTP GET message.

The reason for this is to avoid having the server send back RTCM
data before we are done parsing the HTTP response. If this happens
there will be partial RTCM data along with the HTTP response.

dat-tran-matternet committed:
AVX-64: Update comments explaining the three sections of code

dat-tran-matternet commented on ntripclient.c:
```diff
@@ -845,14 +845,15 @@ int main(int argc, char **argv)
       // this block of code is very long and difficult to read.  I'll attempt
       // to summarize here.
       //
-      // Lines 747 - 1080 handle UDP mode
-      // Lines 1082 - 1433 handle RTSP mode
-      // Lines 1435 - 1822 handle all the other modes
-      // 
```
@rabad-matternet I removed the line numbers so we don't have to keep maintaining them. I changed it to say there are three major sections to look out for.

rabad-matternet replied:
I agree that line numbers are problematic to maintain.  How about putting a comment that can be searched at each section and then state what that searchable comment is here?  

The motivation for the line numbers (and now a searchable comment) is that this function is to big to see what's going on.  So having a way to easily navigate would be nice.

dat-tran-matternet replied:
Done. Great suggestion.

dat-tran-matternet commented:
Will request to pull into dev instead of master.
